master/internal/kubernetes/deletion.go

```diff
@@ -0,0 +1,155 @@
+package kubernetes
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/determined-ai/determined/master/pkg/actor"
+
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	typedV1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// The kubernetesDeletionDealer and kubernetesDeletionWorker actors are responsible for processing
+// the deletion of Kubernetes resources. The single kubernetesDeletionDealer actor receives
+// deletion requests and forwards it to one of the kubernetesDeletionWorker in a round-robin
+// manner.
+//
+// The reason that requests are processed by a pool of workers rather than processed
```

Comment:
blocking: same as the comment in creation. I 100% like the explanation but it would be nicer to explain the message protocol here.